src/database_test.ts

```
@@ -0,0 +1,230 @@
+import mongoose from 'mongoose';
+
+// Define a schema
+const userSchema = new mongoose.Schema({
+    name: String,
+    email: String,
+    password: String,
+    accessLevel: Number
+});
+
+const packageSchema = new mongoose.Schema({
+    name: String,
+    url: String,
+    score: String,
+    version: String,
+    prev_versions: [String]
+});
+// const User = mongoose.model('User', userSchema); // This defines the "users" collection
+const Package = mongoose.model('Package', packageSchema)
+
+export async function connectToMongoDB(database: string) {
+    try {
+        // Replace with your actual MongoDB URI
+        const mongoURI = `mongodb+srv://askannan:IxnNnCuO0ICCZXGl@cluster0.9gpef.mongodb.net/${database}?retryWrites=true&w=majority&appName=Cluster0`;
+
+        // Connect to the MongoDB cluster
+        await mongoose.connect(mongoURI);
+        console.log('Connected to MongoDB');
+    } catch (error) {
+        console.error('Error connecting to MongoDB', error);
+        process.exit(1); // Exit process with failure
+    }
+}
+
+export async function disconnectMongoDB() {
+    try {
+        await mongoose.disconnect();
+        console.log('Disconnected from MongoDB');
+    } catch (error) {
+        console.error('Error disconnecting from MongoDB:', error);
+    }
+}
+
+// might want to make this just go update if it finds that a package with the same name is already present
+export async function addNewPackage(name: String, url: String, score?: String, version?: String, previousVersion?: String) {
+    const newPackage = new Package({
+        name: name,
+        url: url,
+        score: score,
+        version: version,
+        previousVersions: previousVersion
+    });
+
+    try {
+        const savedPackage = await newPackage.save();
+        console.log('Package saved:', savedPackage);
+    } catch (error) {
+        console.error('Error saving package:', error);
```

will need to return an error so that API can return error to frontend to display. please do this for all instances of try catch blocks